src-unix/native/winsize.c

```diff
@@ -10,7 +10,16 @@
 CAMLprim value caml_notty_winsize (value vfd) {
 #ifdef _WIN32
   (void) vfd;
-  caml_failwith("not implemented on windows");
+  HANDLE hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
+  if (hConsole == INVALID_HANDLE_VALUE) return Val_int (0);
+
+  CONSOLE_SCREEN_BUFFER_INFO csbi;
+  if (GetConsoleScreenBufferInfo(hConsole, &csbi)) {
```

Comment:
Here is the doc for this:
https://learn.microsoft.com/en-us/windows/console/getconsolescreenbufferinfo#return-value
https://learn.microsoft.com/en-us/windows/console/console-screen-buffer-info-str
https://learn.microsoft.com/en-us/windows/console/small-rect-str